formver.compiler-plugin/core/src/org/jetbrains/kotlin/formver/core/embeddings/callables/FunctionEmbedding.kt

```diff
@@ -25,3 +31,16 @@ class UserFunctionEmbedding(private val callable: RichCallableEmbedding) : Funct
     override val viperMethod: Method?
         get() = body?.toViperMethod(callable) ?: callable.toViperMethodHeader()
 }
+
+
+/**
+ * An embedding of a user-defined pure function
+ */
+class PureUserFunctionEmbedding(private val callable: RichCallableEmbedding) : PureFunctionEmbedding,
+        CallableEmbedding by callable {
+
+    var body: PureFunctionBodyEmbedding? = null
```

Comment:
Until you need a PureFunctionBody embedding, perhaps just an `ExpEmbedding` would be best here?